pkg/yaml/yaml_test.go

```diff
@@ -27,7 +27,7 @@ func TestReadYamlFile(t *testing.T) {
 	var nonExistingEmptyYamlConf EmptyYamlConfig
 
 	// Setup temporary file
-	file, err := CreateTempFile(t, existingEmptyYamlFileNamePattern)
+	file, err := os.CreateTemp(t.TempDir(), existingEmptyYamlFileNamePattern)
```

Comment:
no need for os.CreateTemp. Just this:
```go
path := filepath.Join(t.TempDir(), "empty.yaml")
```
This also removes the need to close the file, as you're not dealing with a File object anymore.

Reply:
But we need to create the file somehow? I agree that no temporary file is needed but then we need to create it with `os.Create()` because otherwise there is no file I could read.

Reply:
_ = os.WriteFile(filepath.Join(t.TempDir(), "empty.yaml"), nil, 0o600)

Reply:
Done in 0eccf6398618e1200919d26e7cc90505276742d4